### PR TITLE
Update Tenancy Agreement Details Endpoint

### PR DESCRIPTION
**Pull request by humulla**
A new endpoint has been added so that the Tenancy Agreement details can be updated every time any changes are made on MMH. 
Currently, we are only updating the tenancy end date but in the future, we will add more values

Relevant tests have been added to ensure this works as expected

Next step: Trigger this endpoint to the DynamoDB stream on TenureInformation Table

### CODE REVIEW

**Review by Duslerke — COMMENTED** [review]
This may seem like a lot of comments _(some of them will get hidden behind a collapsed view)_,
but a lot of them are related to each other, so there's only a couple of larger things to change. Around one 3rd of the comments are about column name changes.

File: HousingFinanceInterimApi/V1/Gateways/UpdateTAGateway.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        {
+            try
+            {
+                var uhTenancyAgreement = _context.UHTenancyAgreement.SingleOrDefault(p => p.PropRef == query.PropertyReference);
+                var maTenancyAgreement = _context.MATenancyAgreement.SingleOrDefault(p => p.PropRef == query.PropertyReference);
+
+                if (uhTenancyAgreement is not null && maTenancyAgreement is not null)
+                {
+                    uhTenancyAgreement.Eot = request.TenureEndDate;
+                    maTenancyAgreement.Eot = request.TenureEndDate;
```
[the remark this answers is not in the document]
Comment (humulla) [reply HousingFinanceInterimApi/V1/Gateways/UpdateTAGateway.cs:29]: yes, we are currently intending to use this endpoint to only update the tenureEndDate - might be enhanced in the future

File: HousingFinanceInterimApi/V1/Gateways/UpdateTAGateway.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+                        maTenancyAgreement.IsTerminated = true;
+                        maTenancyAgreement.IsPresent = false;
+                    }
+                    else
+                    {
+                        uhTenancyAgreement.IsTerminated = false;
+                        uhTenancyAgreement.IsPresent = true;
+                        maTenancyAgreement.IsTerminated = false;
+                        maTenancyAgreement.IsPresent = true;
+                    }
```
Comment (Duslerke) [inline HousingFinanceInterimApi/V1/Gateways/UpdateTAGateway.cs:44]: I believe I saw this logic being calculated within the UseCase now.

File: HousingFinanceInterimApi/V1/UseCase/UpdateTAUseCase.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+        public UpdateTAUseCase(IUpdateTAGateway gateway)
+        {
+            _gateway = gateway;
+        }
+
+        public async Task ExecuteAsync(string tagRef, UpdateTARequest request)
+        {
+            var convertToString = request.TenureEndDate.ToString();
+            if (request.TenureEndDate == null || request.TenureEndDate > DateTime.UtcNow)
```
Comment (Duslerke) [inline HousingFinanceInterimApi/V1/UseCase/UpdateTAUseCase.cs:21]: If NULL, if in the future, or if 1900s. See the comment on the usecase tests.

**Review by Duslerke — APPROVED** [review]

**Comment by Duslerke** [comment]
Woops, accidentally approved. Must be tired. :sweat: 
2 things that need changing:

1. The 1900s end date means tenancy is active.
2. The Controller still seems to be calling a Gateway directly (maybe some commits weren't pushed up?)

The rest is not functionally significant.

**Review by FaisalHackney — APPROVED** [review]

**Pull request merged**